aggregator/counter_elem.gen.go

```diff
@@ -240,27 +239,24 @@ func (e *CounterElem) Consume(
 		e.toConsume[i].lockedAgg.closed = true
 		e.toConsume[i].lockedAgg.aggregation.Close()
 		if e.toConsume[i].lockedAgg.sourcesSeen != nil {
-			sourceSetSize := len(e.toConsume[i].lockedAgg.sourcesSeen)
-			// If the source set we are about to cache is too big (i.e., bigger than the
-			// configured max size), we do not want to retain it and instead simply discard it.
-			// This is useful if say over the course of a month the metric IDs producing
-			// the forward metric ID have changed significantly so there are a lot of stale
-			// IDs in the map so we regenerate the map once in a while to refresh the map.
-			if sourceSetSize <= e.opts.MaxCachedSourceSetSize() {
-				e.cachedSourceSetsLock.Lock()
-				// This is to make sure there aren't too many cached source sets taking up
-				// too much space.
-				if len(e.cachedSourceSets) < e.opts.MaxNumCachedSourceSets() {
-					e.cachedSourceSets = append(e.cachedSourceSets, e.toConsume[i].lockedAgg.sourcesSeen)
-				}
-				e.cachedSourceSetsLock.Unlock()
+			e.cachedSourceSetsLock.Lock()
+			// This is to make sure there aren't too many cached source sets taking up
+			// too much space.
+			if len(e.cachedSourceSets) < e.opts.MaxNumCachedSourceSets() {
```

Comment:
Will the cached source sets all be about the same size? It seems previously we were considering the total size of the source sets whereas now we only consider the total number of source sets.

Reply:
Each item is a bitset that's storing shard ids, so yeah they are limited in size.